sfdoc/publish/tasks.py

```diff
@@ -10,44 +10,110 @@
 
 from .amazon import S3
 from .exceptions import SfdocError
-from .html import HTML
+from .html import HTML, collect_html_paths
 from .logger import get_logger
 from .models import Article
 from .models import Bundle
 from .models import Image
 from .models import Webhook
-from .salesforce import Salesforce
-from .logger import get_logger
-from .utils import is_html
-from .utils import skip_html_file
-from .utils import unzip
+from .salesforce import SalesforceArticles
+from . import utils
 
 
-def _process_bundle(bundle, path, enforce_no_duplicates = True):
+def _download_and_unpack_easydita_bundle(bundle, path):
     logger = get_logger(bundle)
-    # get APIs
-    salesforce = Salesforce()
-    s3 = S3()
-    # download bundle
+
     logger.info('Downloading easyDITA bundle from %s', bundle.url)
+    assert bundle.url.startswith("https://")
     auth = (settings.EASYDITA_USERNAME, settings.EASYDITA_PASSWORD)
     response = requests.get(bundle.url, auth=auth)
     zip_file = BytesIO(response.content)
-    unzip(zip_file, path, recursive=True)
+    utils.unzip(zip_file, path, recursive=True, ignore_patterns=["*/assets/*"])
+
+    # with TemporaryDirectory(f"bundle_{bundle.pk}_") as tempdir:
+
+    #     rootpath = utils.find_bundle_root_directory(tempdir)
+    #     assert os.path.exists(os.path.join(rootpath, "log.txt"))
+    #     utils.sync_directories(rootpath, path)
+    #     assert os.path.exists(os.path.join(path, "log.txt"))
+
+
+def _process_bundle(bundle, path):
+    logger = get_logger(bundle)
+
+    # get APIs
+    salesforce_docset = SalesforceArticles(bundle.docset_id)
+    s3 = S3(bundle)
+
+    s3.delete_draft_images()
+
+    assert os.path.exists(path)
+
+    # get new files from EasyDITA and put them on top
+    _download_and_unpack_easydita_bundle(bundle, path)
+
+    path = utils.find_bundle_root_directory(path)
+
+    # try to name docset if necessary
+    if not bundle.docset.name:
+        try_name_docset(bundle.docset, path)
+
     # collect paths to all HTML files
-    html_files = []
+    html_files = collect_html_paths(path, logger)
+
+    create_drafts(bundle, html_files, path, salesforce_docset, s3)
+
+
+def try_name_docset(docset, path):
+    """Try to name a docset from information in an index HTML"""
     for dirpath, dirnames, filenames in os.walk(path):
-        for filename in filenames:
-            filename_full = os.path.join(dirpath, filename)
-            if is_html(filename):
-                if skip_html_file(filename):
-                    logger.info(
-                        'Skipping file: %s',
-                        filename_full.replace(path + os.sep, ''),
-                    )
-                    continue
-                html_files.append(filename_full)
-    # check all HTML files and create list of image files
+        html_files = [filename for filename in filenames if ".htm" in filename]
+        if html_files:
+            index_file = os.path.join(dirpath, html_files[0])
```

Comment:
We're just assuming the first file is the index?

Reply:
10ff30aed821ba55fac7d67e7887b035b87b0af3